docs/features/watchdog.md

```diff
@@ -124,26 +136,207 @@ watchdog.on( 'change:state' ( evt, name, currentState, prevState ) => {
+const watchdog = new ContextWatchdog( Context, {
+	crashNumberLimit: 10
+} );
+
+// Initialize the watchdog with context configuration:
+await watchdog.create( {
+	plugins: [
+	    // ...
+	],
+	// ...
+} );
+
+// Add editor instances.
+// You mat also use multiple `ContextWatchdog#add()` calls, each adding a single editor.
+await watchdog.add( [ {
+	id: 'editor1',
+	type: 'editor',
+	sourceElementOrData: document.querySelector( '#editor' ),
+	config: {
+		plugins: [ Essentials, Paragraph, Bold, Italic ],
+		toolbar: [ 'bold', 'italic', 'alignment' ]
+	},
+	creator: ( element, config ) => ClassicEditor.create( element, config )
+}, {
+	id: 'editor2',
+	type: 'editor',
+	sourceElementOrData: document.querySelector( '#editor' ),
+	config: {
+		plugins: [ Essentials, Paragraph, Bold, Italic ],
+		toolbar: [ 'bold', 'italic', 'alignment' ]
+	},
+	creator: ( element, config ) => ClassicEditor.create( element, config )
+} ] );
+
+// Or:
+await watchdog.add( {
+	id: 'editor1',
+	type: 'editor',
+	sourceElementOrData: document.querySelector( '#editor' ),
+	config: {
+		plugins: [ Essentials, Paragraph, Bold, Italic ],
+		toolbar: [ 'bold', 'italic', 'alignment' ]
+	},
+	creator: ( element, config ) => ClassicEditor.create( element, config )
+} );
+
+await watchdog.add( {
+    id: 'editor2',
+    type: 'editor',
+    sourceElementOrData: document.querySelector( '#editor' ),
+    config: {
+        plugins: [ Essentials, Paragraph, Bold, Italic ],
+        toolbar: [ 'bold', 'italic', 'alignment' ]
+    },
+    creator: ( element, config ) => ClassicEditor.create( element, config )
+} );
+```
+
+To destroy one of the item instances use {@link module:watchdog/contextwatchdog~ContextWatchdog#remove `ContextWatchdog#remove`}:
+
+```js
+await watchdog.remove( [ 'editor1', 'editor2' ] );
+
+// Or:
+await watchdog.remove( 'editor1' );
+await watchdog.remove( 'editor2' );
+```
+
+<info-box>
+	Examples presents the "synchronous way" of the integration with the context watchdog feature, however it's not needed to wait for the promises returned by the `create()`, `add()` and `remove()` methods. There might be a need
+	to create and destroy items dynamically with shared context and that can be easily achieved as all promises operating on the internal API will be chained.
+</info-box>
+
+#### Context watchdog API
+
+The context watchdog feature provides the following API:
+
+```js
+// Creating watchdog that will use the `Context` class and watchdog configuration.
+const watchdog = new ContextWatchdog( Context, watchdogConfig );
+
+// Setting a custom creator for the context.
+watchdog.setCreator( async config => {
+	const context = await Context.create( config );
+
+	// Do something when the context is initialized.
+
+	return context;
+} );
+
+// Setting a custom destructor for the context.
+watchdog.setDestructor( async context => {
+	// Do something before destroy.
+
+	await context.destroy();
```

Comment:
```suggestion
	return await context.destroy();
```

Reply:
You don't have to do that. Since this is a async function it already returns a promise that awaits for all \`await somePromise;\` things.

Reply:
:+1: